tests/unit/test_project.py

```diff
@@ -592,3 +588,52 @@ def test_project_generate_metadata(yaml_loaded_data):
 def test_project_yaml(yaml_loaded_data):
     project = Project.unmarshal(yaml_loaded_data)
     assert project.to_yaml() == EXPECTED_DUMPED_YAML
+
+
+@pytest.mark.parametrize(
+    ["platforms", "expected_build_infos"],
+    [
+        (
+            {
+                "amd64": None,
+            },
+            [
+                RockcraftBuildInfo(
+                    build_on="amd64",
+                    build_for="amd64",
+                    base=BaseName(name="ubuntu", version="20.04"),
+                    platform_entry="amd64",
+                    build_for_variant=None,
+                )
+            ],
+        ),
+        (
+            {
+                "amd64": {
+                    "build-on": ["amd64", "i386"],
+                    "build-for": ["amd64"],
+                },
+            },
+            [
+                RockcraftBuildInfo(
+                    build_on="amd64",
+                    build_for="amd64",
+                    base=BaseName(name="ubuntu", version="20.04"),
+                    platform_entry="amd64",
+                    build_for_variant=None,
+                ),
+                RockcraftBuildInfo(
+                    build_on="i386",
+                    build_for="amd64",
+                    base=BaseName(name="ubuntu", version="20.04"),
+                    platform_entry="amd64",
+                    build_for_variant=None,
+                ),
+            ],
+        ),
+    ],
+)
+def test_project_get_build_plan(yaml_loaded_data, platforms, expected_build_infos):
```

Comment:
Maybe also test with platform names that don't match an architecture name?

Reply:
oh of course! Done